src/html5_parser/node.rs

```diff
@@ -212,6 +219,38 @@ impl Node {
         false
     }
 
+    /// Check if node has a named ID
+    pub fn has_named_id(&self) -> bool {
+        if self.type_of() != NodeType::Element {
+            return false;
+        }
+
+        self.named_id.is_some()
+    }
+
+    /// Set named ID (only applies to Element type, does nothing otherwise)
+    pub fn set_named_id(&mut self, named_id: &str) {
+        if self.type_of() == NodeType::Element {
+            self.named_id = Some(named_id.to_owned());
+            // TODO: log a warning/error if this fails for some reason
+            let _ = self.insert_attribute("id", named_id);
+        }
```

Comment:
Looking at the implementation of `self.insert_attribute`, I see that the function could be infallible (never return an error) if we never failed the [type check at the top of the function](https://github.com/gosub-browser/gosub-engine/blob/6fb63ba7083b0280224e84b33b7560ca7af260a4/src/html5_parser/node.rs#L273-L275).  Then I realized that we're doing that type check in a number of functions.  That tells me we're using a single struct where we should be using several.  One struct doesn't have these functions defined, and another has them defined but no `self.type_of()` test.

Reply:
I'm not exactly sure how to split these structs apart, yet, though, so I don't have a concrete suggestion at this point.

Reply:
Yes, I had the same thought, but wasn't sure at the moment how we wanted to split them apart. The multiple type checking is a bit annoying but I would like to break those apart at some point